packages/next/server/next-server.ts

```diff
@@ -2048,29 +2046,28 @@ export default class NextNodeServer extends BaseServer {
     }
 
     // For middleware to "fetch" we must always provide an absolute URL
-    const isDataReq = !!params.query.__nextDataReq
-    const query = urlQueryToSearchParams(params.query).toString()
-    const locale = params.query.__nextLocale
-    // Use original pathname (without `/page`) instead of appPath for url
-    let normalizedPathname = params.page
+    const locale = query.__nextLocale
+    const isDataReq = !!query.__nextDataReq
+    const queryString = urlQueryToSearchParams(query).toString()
 
     if (isDataReq) {
       params.req.headers['x-nextjs-data'] = '1'
     }
 
+    let normalizedPathname = params.page
     if (isDynamicRoute(normalizedPathname)) {
       const routeRegex = getNamedRouteRegex(params.page)
       normalizedPathname = interpolateDynamicPath(
         params.page,
-        Object.assign({}, params.params, params.query),
+        Object.assign({}, params.params, query),
         routeRegex
       )
     }
 
     const url = `${getRequestMeta(params.req, '_protocol')}://${
       this.hostname
-    }:${this.port}${locale ? `/${locale}` : ''}${normalizedPathname}${
-      query ? `?${query}` : ''
+    }:${this.port}${locale ? `/${locale}` : ''}${pathname}${
```

Comment:
```suggestion
    }:${this.port}${locale ? `/${locale}` : ''}${pathname}${
```
For the rewrite case is the `pathname` here the URL before rewrite or the destination after rewrite?

If before resolving the rewrite this breaks resolving the page in the `web-server` iirc which is why `normalizedPathname` was used here